src/controllers/hid/hidcontroller.cpp

```diff
@@ -293,8 +293,32 @@ void HidController::sendBytesReport(QByteArray data, unsigned int reportID) {
         }
     } else {
         controllerDebug(result << "bytes sent to" << getName()
-                 << "serial #" << hid_serial
-                 << "(including report ID of" << reportID << ")");
+                               << "serial #" << hid_serial
+                               << "(including report ID of" << reportID << ")");
+    }
+}
+
+void HidController::sendFeatureReport(
+        const QList<int>& dataList, unsigned int reportID) {
+    QByteArray dataArray;
+    for (const int datum : dataList) {
+        dataArray.append(datum);
+    }
+
+    // Append the Report ID to the beginning of dataArray[] per the API..
+    dataArray.prepend(reportID);
+
+    int result = hid_send_feature_report(m_pHidDevice,
+            reinterpret_cast<const unsigned char*>(dataArray.constData()),
+            dataArray.size());
+    if (result == -1) {
+        qWarning() << "sendFeatureReport is unable to send data to" << getName()
+                   << "serial #" << hid_serial << ":"
+                   << safeDecodeWideString(hid_error(m_pHidDevice), 512);
```

Comment:
Can we make 512 a constexpr in the anonymouse namespace?

Reply:
Sorry, I don't know how to do this. Isn't this the length of an error message, which isn't known at compile time?

Reply:
The literal `512` is used repeatedly in the HID code. I suggest to leave it as is to not complicate this PR. Should be fixed when refactoring HidController.

@Holzhaus The `max_length` parameter is always set to 512 so safeDecodeWideString() doesn't need to not expose it. It could later be used as a default value for this parameter.

Reply:
512 replaced by a constexpr now